internal/generate/generate.go

```diff
@@ -8,26 +8,33 @@ import (
 	"time"
 )
 
-const alphanum = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
+const (
+	CharsetAlphaNumeric = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
+	CharsetPassword     = CharsetAlphaNumeric + `!@#$%^&*()_+-=[]|;:,./<>?`
+)
 
 func init() {
 	mathrand.Seed(time.Now().UnixNano())
 }
 
-// CryptoRandom generates a cryptographically-safe random number
-func CryptoRandom(n int) (string, error) {
+// CryptoRandom generates a cryptographically-safe random number. defaults to alphanumeric charset.
+func CryptoRandom(n int, charset ...string) (string, error) {
 	if n <= 0 {
 		return "", nil
 	}
+	useCharset := CharsetAlphaNumeric
+	if len(charset) > 0 {
+		useCharset = charset[0]
```

Comment:
Using a var args as an option argument is really error prone. The caller has no way to know that all the other arguments are going to be ignored.

I think we should either create two functions (one that accepts the charset and one that does not), or simply require that all callers specify the charset, which would be my preference.

There are only 6 other callers (2 in production code, 4 in tests) that don't pass a charset, so it should be easy to update all of them.

Another option could be to actually use all the charsets by appending them together. That seems fine as well.